derp/derp_server.go

```diff
@@ -182,6 +186,12 @@ func NewServer(privateKey key.Private, logf logger.Logf) *Server {
 		memSys0:              ms.Sys,
 		watchers:             map[*sclient]bool{},
 		sentTo:               map[key.Public]map[key.Public]int64{},
+		bytes_pool: &sync.Pool{
+			New: func() interface{} {
+        var v []byte
+        return v
```

Comment:
a slice (three words) allocates when put through an `interface{}`, so []byte pools use `*[]byte` instead.

Reply:
yep, staticcheck pointed this out as well, but when I changed it over to a `*[]byte`, it seemed to perform worse. I'm not sure if this is an issue with how I implemented it or not, but if there's no issue would it make sense to keep it as `[]byte` then?

Reply:
No, we really don't want to pay the cost of shoving a slice into an interface in code that's trying to prevent allocations.

If it performs worse, you need to understand why. You can use pprof to see where the CPU & allocs are going. That's integrated nicely into the `go` tool when you write Benchmark funcs.

Reply:
I think I was too aggressively putting stuff back into the pool, causing extra allocations of shared pointers possibly